agent/acs/handler/payload_responder.go

```diff
@@ -0,0 +1,289 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handler
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	apiappmesh "github.com/aws/amazon-ecs-agent/agent/api/appmesh"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	apitaskstatus "github.com/aws/amazon-ecs-agent/agent/api/task/status"
+	"github.com/aws/amazon-ecs-agent/agent/data"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/agent/eventhandler"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	apieni "github.com/aws/amazon-ecs-agent/ecs-agent/api/eni"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/credentials"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	loggerfield "github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+// skipAddTaskComparatorFunc defines the function pointer that accepts task status
+// and returns the boolean comparison result.
+type skipAddTaskComparatorFunc func(apitaskstatus.TaskStatus) bool
+
+// taskMessageHandler implements TaskMessageHandler.
+type taskMessageHandler struct {
+	taskEngine                  engine.TaskEngine
+	ecsClient                   api.ECSClient
+	dataClient                  data.Client
+	taskHandler                 *eventhandler.TaskHandler
+	credentialsManager          credentials.Manager
+	latestSeqNumberTaskManifest *int64
+}
+
+func (tmHandler *taskMessageHandler) ProcessTaskMessage(message *ecsacs.PayloadMessage,
+	ackFunc func(*ecsacs.AckRequest, []*ecsacs.IAMRoleCredentialsAckRequest)) error {
+
+	credentialsAcks, allTasksHandled := tmHandler.addPayloadTasks(message)
+
+	// Update latestSeqNumberTaskManifest for it to get updated in state file.
+	if tmHandler.latestSeqNumberTaskManifest != nil && message.SeqNum != nil &&
+		*tmHandler.latestSeqNumberTaskManifest < *message.SeqNum {
+		*tmHandler.latestSeqNumberTaskManifest = *message.SeqNum
+	}
+
+	if !allTasksHandled {
+		return fmt.Errorf("did not handle all tasks")
```

Comment:
Maybe naive but why are we not using errors.Errorf() here?

Reply:
I don't believe there's any particular reason `fmt.Errorf()` is used here over `errors.Errorf()` other than that `fmt.Errorf()` is what is currently used in the existing code for this case. Both functions return an error.

If you have a strong opinion on this but it is non-blocking, I can potentially make future changes to use `errors.Errorf()` in favor of `fmt.Errorf()` wherever we are using it but I feel that should be left out of scope of this specific PR. 

Let me know what you think, thanks!

Reply:
Sure sounds good to me. Maybe add it in TODOs for the PR for the future reference. Thanks.